server/sonar-server/src/main/java/org/sonar/server/platform/RoutesFilter.java

```diff
@@ -17,60 +17,46 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
+package org.sonar.server.platform;
 
-package org.sonar.server.authentication;
-
-import static org.sonar.api.web.ServletFilter.UrlPattern.Builder.staticResourcePatterns;
+import static java.lang.String.format;
 
 import java.io.IOException;
+import javax.servlet.Filter;
 import javax.servlet.FilterChain;
 import javax.servlet.FilterConfig;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import org.sonar.api.server.ServerSide;
-import org.sonar.api.web.ServletFilter;
-import org.sonar.server.exceptions.UnauthorizedException;
-
-@ServerSide
-public class ValidateJwtTokenFilter extends ServletFilter {
 
-  private final JwtHttpHandler jwtHttpHandler;
-
-  public ValidateJwtTokenFilter(JwtHttpHandler jwtHttpHandler) {
-    this.jwtHttpHandler = jwtHttpHandler;
-  }
+public class RoutesFilter implements Filter {
 
-  @Override
-  public UrlPattern doGetPattern() {
-    return UrlPattern.builder()
-      .includes("/*")
-      .excludes(staticResourcePatterns())
-      .build();
-  }
+  private static final String BATCH_CONTROLLER = "/batch/";
+  private static final String EMPTY = "";
 
   @Override
   public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
     HttpServletRequest request = (HttpServletRequest) servletRequest;
     HttpServletResponse response = (HttpServletResponse) servletResponse;
-
-    try {
-      jwtHttpHandler.validateToken(request, response);
+    String path = request.getRequestURI().replaceFirst(request.getContextPath(), EMPTY);
+    if (path.startsWith(BATCH_CONTROLLER) && path.endsWith(".jar")) {
+      response.sendRedirect(format("%s%sfile?name=%s", request.getContextPath(), BATCH_CONTROLLER, path.replace(BATCH_CONTROLLER, EMPTY)));
+    } else if ("/batch_bootstrap/index".equals(path)) {
+      response.sendRedirect(format("%s%sindex", request.getContextPath(), BATCH_CONTROLLER));
+    } else {
       chain.doFilter(request, response);
-    } catch (UnauthorizedException e) {
-      response.setStatus(e.httpCode());
     }
   }
 
   @Override
   public void init(FilterConfig filterConfig) throws ServletException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1186)